src/pyodide/internal/workers.py

```diff
@@ -682,6 +719,138 @@ async def text(self) -> str:
         return await self.js_object.text()
 
 
+def serialize_to_python(obj: "JsProxy"):
+    """
+    Converts JS objects like Response, Request, Blob, etc. to equivalent Python objects defined in
+    this module and also other JS objects like Map, Set, etc. to equivalent Python stdlib objects.
+
+    This method is used for Workers RPC in Python to convert JavaScript objects to Python. As such
+    it does not support serializing all JS object types.
+    """
+
+    def convert(obj, convert, cache):
```

Comment:
This shadows `obj` and doesn't close over anything, why not pull it out to a top level definition?

Reply:
Because it's only used by `serialize_to_python`, I can rename `obj` so it's not shadowed. I think keeping it inside this method makes the code easier to reason about though.

Reply:
What about moving it to top level as `serialize_to_python_default_converter`? I don't see why it's easier to reason about it when it's a nested function. I guess:

Pros of nested functions:
* clear that it isn't used anywhere else
* definition won't get separated by other functions from the function that uses it.

Cons of nested functions:
* unclear whether it might close over variables
* extra indentation

I only have to worry about whether there are other call sites to a top level function when I want to change it. Whereas I find myself very frequently checking what variables a function is closing over just to understand what the code means. So I think the readability disadvantages of nested functions outweigh the benefits.

Reply:
FWIW nested functions aren't the only ones that can access variables outside of their scope, non-nested functions can access global variables too. So I don't really see that as a unique concern for nested functions.

It would be nice if Python had the ability to force nested functions not to close over any args.

But alright, I'm not that strongly opposed to moving this into its own top-level function, so I'll just do it.